server/routes/robots.js

```diff
@@ -0,0 +1,31 @@
+import app from 'server/server';
+import { handleErrors } from 'server/utils/errors';
+import { getInitialData } from 'server/utils/initData';
+import { hostIsValid } from 'server/utils/routes';
+import { communityUrl } from 'utils/canonicalUrls';
+
+const BASE_ROBOTS = `User-agent: *
+Disallow:`;
+
+const buildRobotsFile = (community) => {
+	if (community) {
+		return `${BASE_ROBOTS}
+Sitemap: ${communityUrl(community)}/sitemap-index.xml`;
+	}
+	return BASE_ROBOTS;
+};
+
+app.get('/robots.txt', async (req, res, next) => {
+	if (!hostIsValid(req, 'community')) {
+		return buildRobotsFile();
+	}
+
+	try {
```

Comment:
We have a mysterious function called `wrap` in server.js that lets you write async Express routes without having to do this error handling by hand. Hopefully the library should be able to handle this by itself soon...

Reply:
Yeah I saw it mostly used in `api` files so I didn't know if it was only idiomatic to use it there. I'll use it here!